### DISCUSSION TITLE
v3.8.2

### DISCUSSION THREAD

**Discussion by markwhitfeld** [ngxs/store#2086] [Releases]
## What's Changed
* build: add angular 17 support by @jpraetorius in https://github.com/ngxs/store/pull/2079
* docs: update `options.md` by @arturovt in https://github.com/ngxs/store/pull/2023
* refactor(store): remove `isAngularInTestMode` by @arturovt in https://github.com/ngxs/store/pull/2024
* refactor(store): do not create new functions in each `@State` decorator by @arturovt in https://github.com/ngxs/store/pull/2022
* refactor(store): do not create new functions per each new context by @arturovt in https://github.com/ngxs/store/pull/2025
* perf(store): tree-shake selectors validation errors by @arturovt in https://github.com/ngxs/store/pull/2020
* docs: fix upsertItem operator example by @nilzen in https://github.com/ngxs/store/pull/2028
* fix(store): patch operator must handle existing nulls by @TheKingDave in https://github.com/ngxs/store/pull/2064
* fix(devtools-plugin): do not re-enter Angular zone when resetting state by @arturovt in https://github.com/ngxs/store/pull/2038
* fix(storage-plugin): access local and session storages globals only in browser by @arturovt in https://github.com/ngxs/store/pull/2034
* fix(storage-plugin): require only `getItem` and `setItem` on engines by @arturovt in https://github.com/ngxs/store/pull/2036
* ci: update node images and browser tools by @arturovt in https://github.com/ngxs/store/pull/2039
* refactor(*): replace `get type()` with `type =` in actions by @arturovt in https://github.com/ngxs/store/pull/2035

## New Contributors
* @jpraetorius made their first contribution in https://github.com/ngxs/store/pull/2079
* @TheKingDave made their first contribution in https://github.com/ngxs/store/pull/2064
* @nilzen made their first contribution in https://github.com/ngxs/store/pull/2028

**Full Changelog**: https://github.com/ngxs/store/compare/v3.8.1...v3.8.2

<hr /><em>This discussion was created from the release <a href='https://github.com/ngxs/store/releases/tag/v3.8.2'>v3.8.2</a>.</em>

**Comment by hakimio** [comment]
What happened to **3.9** release #2065?

**Reply by markwhitfeld** [reply]
It is still in process. There are some additional PRs that we would like to include in that release.
We released this version with some bug fixes, as well as the angular version upgrade so that users are not blocked with upgrading angular.

**Reply by kiruthin** [reply]
@markwhitfeld  Thanks Mark. Is there any way for us to try out a beta version of 3.9? Appreciate all the work.
